pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator.go

```diff
@@ -114,6 +115,32 @@ func (mutator *VMIsMutator) Mutate(ar *v1beta1.AdmissionReview) *v1beta1.Admissi
 	}
 }
 
+func (mutator *VMIsMutator) setDefaultNetworkInterface(obj *v1.VirtualMachineInstance) {
+	autoAttach := obj.Spec.Domain.Devices.AutoattachPodInterface
+	if autoAttach != nil && *autoAttach == false {
+		return
+	}
+
+	// Override only when nothing is specified
+	if len(obj.Spec.Networks) == 0 {
+		iface := mutator.ClusterConfig.GetDefaultNetworkInterface()
+		switch iface {
+		case "bridge":
+			obj.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultBridgeNetworkInterface()}
+		case "masquerade":
+			obj.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultMasqueradeNetworkInterface()}
+		case "slirp":
+			defaultIface := v1.DefaultSlirpNetworkInterface()
+			if mutator.ClusterConfig.IsSlirpInterfaceDisabled() {
+				defaultIface = v1.DefaultMasqueradeNetworkInterface()
```

Comment:
I think we should fail instead.

Reply:
Ok. Yea, I wasn't sure what's best to do here.

Reply:
This should now be rejected by a validating webhook. I've also added a functional test.